controllers/config_map.go

```diff
@@ -47,70 +49,142 @@ type ConfigMapReconciler struct {
 	AssumeRole   string
 }
 
+type ClusterInfo struct {
+	Name           string
+	Namespace      string
+	Region         string
+	AWSPartition   string
+	VpcID          string
+	RoleArn        arn.ARN
+	SecurityGroups *crossplaneConfigValuesAWSClusterSecurityGroups
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *ConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&capa.AWSCluster{}).
+		For(&capi.Cluster{}).
 		Complete(r)
 }
 
 func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx)
 
-	cluster := &capa.AWSCluster{}
-	err := r.Client.Get(ctx, req.NamespacedName, cluster)
+	awsCluster := &capa.AWSCluster{}
+	ekscontrolplane := &eks.AWSManagedControlPlane{}
+	clusterInfo := &ClusterInfo{}
+
+	capiCluster := &capi.Cluster{}
+	err := r.Client.Get(ctx, req.NamespacedName, capiCluster)
+
 	if err != nil {
 		logger.Error(err, "failed to get cluster")
 		return ctrl.Result{}, errors.WithStack(client.IgnoreNotFound(err))
 	}
 
-	if !cluster.DeletionTimestamp.IsZero() {
+	if !capiCluster.DeletionTimestamp.IsZero() {
 		logger.Info("Reconciling delete")
-		return r.reconcileDelete(ctx, cluster)
+		return r.reconcileDelete(ctx, capiCluster)
+	}
+
+	err = r.Client.Get(ctx, req.NamespacedName, awsCluster)
```

Comment:
I think it's not best  to try to get AWSCluster and if it doesn't exist try to get AWSManagedControlPlane, I think it would be better to check Cluster values and made a decision based on the referenced values
for example like here:
https://github.com/giantswarm/cluster-apps-operator/blob/main/service/controller/key/key.go#L86-L91



`cluster.Spec.InfrastructureRef.Kind` is `AWSCluster` for CAPA and `AWSManagedCluster` for EKS

With such code its super easy to read what is the code doing  and understand  the behavior